config/config.go

```diff
@@ -530,10 +532,69 @@ type AuthConfig struct {
 	TLSAuthEntities []string            `json:"tls_auth_entities"`
 }
 
+// WebOauthConfig contains the structued AuthHandlerConfig for the CredentialsTypeOAuthWeb type.
+type WebOauthConfig struct {
+	// The allowed jwt audiences the OAuthWeb auth handler will accept.
+	AllowedAudiences []string `json:"allowed_audiences"`
+	// contains the raw jwks json.
+	JSONKeySet AttributeMap `json:"jwks"`
+
+	// on validation the JSONKeySet is validated and parsed into this field and can be used.
+	ValidatedKeySet jwks.KeySet `json:"-"`
+}
+
+// Validate returns true if the WebOAuth is valid. Ensures each key is valid and meets the required constraints.
+func (c *WebOauthConfig) Validate(path string) error {
+	jsonJWKs, err := json.Marshal(c.JSONKeySet)
+	if err != nil {
+		return utils.NewConfigValidationError(fmt.Sprintf("%s.config", path), errors.Wrap(err, "failed to marshal jwks"))
+	}
+
+	keyset, err := jwks.ParseKeySet(string(jsonJWKs))
+	if err != nil {
+		return utils.NewConfigValidationError(fmt.Sprintf("%s.config", path), errors.Wrap(err, "failed to parse jwks"))
+	}
+
+	allowedKeyTypes := map[string]bool{
+		"RSA": true,
+	}
+
+	allowedAlgs := map[string]bool{
+		"RS256": true,
+		"RS384": true,
+		"RS512": true,
+	}
+
+	for i := 0; i < keyset.Len(); i++ {
```

Comment:
we should validate it's non-empty for security reasons

Reply:
Done. Though an empty list does not provide any security concerns. It defaults to closed.